plugin/pkg/shoot/validator/admission.go

```diff
@@ -949,8 +953,12 @@ func validateMachineImagesConstraints(constraints []garden.MachineImage, image,
 	validValues := []string{}
 
 	for _, v := range constraints {
+		if v.ExpirationDate != nil && v.ExpirationDate.Time.UTC().Before(time.Now().UTC()) {
+			continue
+		}
+
 		validValues = append(validValues, fmt.Sprintf("%+v", v))
-		if apiequality.Semantic.DeepEqual(v, *image) {
+		if v.Name == image.Name && v.Version == image.Version {
 			return true, nil
```

Comment:
Here, you should check that the expiration time (if exists) of selected image `v` is not expired.

Reply:
see above